Sources/ClientRuntime/Config/ContextBuilder+withSmithyDefaultConfig.swift

```diff
@@ -0,0 +1,27 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import class Smithy.ContextBuilder
+
+extension ContextBuilder {
+    public typealias SmithyConfig = DefaultClientConfiguration & DefaultHttpClientConfiguration
+
+    public func withSmithyDefaultConfig<Config: SmithyConfig>(
+        _ config: Config
+    ) -> Smithy.ContextBuilder {
+        return self
+            .withIdempotencyTokenGenerator(value: config.idempotencyTokenGenerator)
+            .withLogger(value: config.logger)
+            .withPartitionID(value: config.partitionID)
+            .withAuthSchemes(value: config.authSchemes ?? [])
+            .withAuthSchemePreference(value: config.authSchemePreference)
+            .withAuthSchemeResolver(value: config.authSchemeResolver)
+            .withSocketTimeout(value: config.httpClientConfiguration.socketTimeout)
+            .withIdentityResolver(value: config.bearerTokenIdentityResolver, schemeID: "smithy.api#httpBearerAuth")
+            .withIdentityResolver(value: config.awsCredentialIdentityResolver, schemeID: "aws.auth#sigv4")
```

Comment:
The sigv4 identity resolver was previously set in aws-sdk-swift.  Moved here since sigv4 is considered Smithy-native.

Sigv4a (and other AWS-specific auth schemes) still have their identity resolver set in AWS.